src/DotNetty.Codecs/LineBasedFrameDecoder.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs
+{
+    using System;
+    using System.Collections.Generic;
+    using DotNetty.Buffers;
+    using DotNetty.Transport.Channels;
+    using DotNetty.Common.Utilities;
+
+    /// <summary>
+    ///  A decoder that splits the received {@link ByteBuf}s on line endings.
+    ///  Both {@code "\n"} and {@code "\r\n"} are handled.
+    ///  For a more general delimiter-based decoder, see {@link DelimiterBasedFrameDecoder}.
+    /// </summary>
+    public class LineBasedFrameDecoder : ByteToMessageDecoder
+    {
+        /** Maximum length of a frame we're willing to decode.  */
+        readonly int maxLength;
+        /** Whether or not to throw an exception as soon as we exceed maxLength. */
```

Comment:
Sorry, I haven't realize, what is the comment I need to change and how? :) I have to change all {@code} and {@link} in summary block?